Lux/src/lux_web.cpp

```diff
@@ -797,8 +800,567 @@ std::string get_recording_state() {
     }
 }
 
+// Camera-specific functions for live processing
+bool add_camera_frame(val image_data, int width, int height, std::string temp_name = "live_camera") {
+    try {
+        // Create a temporary image for camera frame
+        auto temp_img_ptr = std::make_unique<uimage>(vec2i{width, height});
+        if (!temp_img_ptr) {
+            std::cerr << "Failed to create camera frame image" << std::endl;
+            return false;
+        }
+
+        uimage temp_img = *temp_img_ptr;
+        unsigned char* img_buffer = (unsigned char*)temp_img.get_base_ptr();
+        size_t buffer_length = width * height * 4;
+
+        // Copy the data from the TypedArray (camera frame)
+        for (unsigned int i = 0; i < buffer_length; ++i) {
+            img_buffer[i] = image_data[i].as<uint8_t>();
+        }
+
+        // Add to scene buffers as temporary camera source
+        ubuf_ptr camera_buf(new buffer_pair<ucolor>(temp_img));
+        any_buffer_pair_ptr any_buf = camera_buf;
+        global_context->s->buffers[temp_name] = any_buf;
+
+        // Mark for re-render
+        global_context->s->ui.displayed = false;
+        
+        return true;
+    } catch (const std::exception& e) {
+        std::cerr << "Error in add_camera_frame: " << e.what() << std::endl;
+        return false;
+    }
+}
+
+bool process_live_camera_frame(val image_data, int width, int height) {
+    // Process camera frame for live preview with effects
+    if (!add_camera_frame(image_data, width, height, "live_camera_preview")) {
+        return false;
+    }
+    
+    // Update source to camera preview for live effects
+    try {
+        if (global_context->s->functions.count("source_image_menu")) {
```

Comment:
Here the menu is hard coded as `source_image_menu` but really any number of menus within the scene should be able to access live camera. Instead, maybe give live camera buffer a name within the scene namespace and then if image picker sees this name it shows live camera icon.